Develop dual fix dual mode

haiwentelink opened the pull request.

andriy-bilynskyy commented:
The following west manifest projects have been modified in this Pull Request:

| Name | Old Revision | New Revision | Diff |
| ---- | ------------ | ------------ |------|
| hal_telink | https://github.com/telink-semi/hal_telink/commit/54c4302c9038d2cf412348400bcd039f5674da91 (develop) | https://github.com/telink-semi/hal_telink/commit/2a999d970883ce294dc1c60479b9ddc90b880bfe (develop_dual) | [telink-semi/hal_telink@54c4302c..2a999d97](https://github.com/telink-semi/hal_telink/compare/54c4302c9038d2cf412348400bcd039f5674da91..2a999d970883ce294dc1c60479b9ddc90b880bfe) |
| mcuboot | https://github.com/telink-semi/mcuboot/commit/1558e7ab0aadb4eac11f03befb5ccd3fa3f0aafe | https://github.com/telink-semi/mcuboot/commit/a07bc815e9484a8921e115f94d4f1121585494ce (develop_dual) | [telink-semi/mcuboot@1558e7ab..a07bc815](https://github.com/telink-semi/mcuboot/compare/1558e7ab0aadb4eac11f03befb5ccd3fa3f0aafe..a07bc815e9484a8921e115f94d4f1121585494ce) |

*Note: This message is automatically posted and updated by the Manifest GitHub Action.*

liucoldstar commented on drivers/i2c/i2c_b9x.c:
```diff
@@ -53,7 +53,7 @@ static int i2c_b9x_configure(const struct device *dev, uint32_t dev_config)
 
 	case I2C_SPEED_FAST:
-		i2c_speed = 400000U;
+		i2c_speed = 200000U;
```
I2C标准定义Fast模式的400k。这个改动的需求是什么？

haiwentelink replied:
already use define CONFIG_I2C_LED

liucoldstar commented on drivers/i2c/i2c_b9x.c:
```diff
@@ -101,7 +101,7 @@ static int i2c_b9x_transfer(const struct device *dev,
 		if (msgs[i].flags & I2C_MSG_READ) {
```
read和write操作对i2c地址的定义有冲突。

haiwentelink replied:
use define CONFIG_I2C_LED

liucoldstar commented on boards/riscv/tl321x/tl3218x-common.dtsi:
```diff
@@ -212,7 +212,7 @@
 &pwm0 {
 	status = "okay";
-	clock-frequency = <300000>;
+	clock-frequency = <10000>;
```
这个修改是基于客户需求做的吗？

haiwentelink replied:
change it back .

liucoldstar commented on kernel/init.c:
```diff
@@ -540,6 +540,17 @@ FUNC_NORETURN void z_cstart(void)
 	z_sys_init_run_level(INIT_LEVEL_PRE_KERNEL_1);
 	z_sys_init_run_level(INIT_LEVEL_PRE_KERNEL_2);
 
+#if CONFIG_STARTUP_OPTIMIZATE
```
建议改成CONFIG_STARTUP_OPTIMIZATION

haiwentelink replied:
changed.

liucoldstar commented on kernel/Kconfig:
```diff
@@ -961,4 +961,10 @@ config HAS_DYNAMIC_DEVICE_HANDLES
 
 endmenu
 
+config STARTUP_OPTIMIZATE
```
建议改成下面的内容：
config STARTUP_OPTIMIZATION
    bool "Optimize startup procedure"
    default n
    help
      The option enables kernel-level optimizations to improve the efficiency of the startup process.

haiwentelink replied:
changed.

haiwentelink committed:
riscv: telink: add the dual mode proc.

- update west.yml for CI and clean code.
- add interface of chipid .
- change the frequency to 96M.
- add early wake proc for customer.
- enable gpioa interface.
- adjust uart inerface.
- update i2c interface for led module.
- add pull-up for the i2c interface.
- disable pwm div part ,or it will overflow.

Signed-off-by: Haiwen Xia <haiwen.xia@telink-semi.com>
